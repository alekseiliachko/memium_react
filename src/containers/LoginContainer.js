import LoginForm from "../components/LoginForm";
import { connect } from "react-redux";
import { loginAttempt } from "../redux/auth/actions";

const mapStateToProps = (store) => ({
  error: store.authReducer.error,
});

const mapDispatchToProps = (dispatch) => ({
  onLogin: (userInfo) => dispatch(loginAttempt(userInfo)),
});

export default connect(mapStateToProps, mapDispatchToProps)(LoginForm);
