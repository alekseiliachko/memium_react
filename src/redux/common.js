export const LOADING_STATUS = {
  NOT_LOADED: "NOT_LOADED",
  LOADING: "LOADING",
  LOADED: "LOADED",
};
