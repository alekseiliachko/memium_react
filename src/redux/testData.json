{
  "authReducer": {
    "username": "alex",
    "authState": "SUCCESS"
  },
  "userReducer": {
    "avatar": "blob:http://localhost:3000/3dcb8dbb-23c7-4abf-8958-52663e0013da",
    "blackList": [
      {
        "accountId": "32df86b7-872d-45d5-8a51-e217f0bc973d",
        "username": "bob",
        "name": null,
        "bio": "populated",
        "imageData": null
      }
    ],
    "details": {
      "accountId": "5d48bb9e-f1d9-41d2-bb02-9c4f578bd282",
      "name": "Круто чел",
      "bio": "populated",
      "gender": "gender",
      "dob": 1615046987793
    },
    "subs": [
      {
        "accountId": "22bb3fe5-185c-4c87-8633-1586586506b7",
        "username": "zilber",
        "name": null,
        "bio": "populated",
        "imageData": null
      }
    ],
    "likedList": [
      {
        "id": "9064e587-3c29-4547-a5a6-51a8149456f8",
        "imageUrl": null,
        "authorId": "5d48bb9e-f1d9-41d2-bb02-9c4f578bd282",
        "title": "Русские мемчанские",
        "description": null,
        "date": 1615046988090,
        "category": "Anime"
      },
      {
        "id": "d3268ef9-bb52-4967-b1ee-14826b9d8a18",
        "imageUrl": null,
        "authorId": "22bb3fe5-185c-4c87-8633-1586586506b7",
        "title": "Chem?",
        "description": null,
        "date": 1615046988090,
        "category": "Chemistry"
      },
      {
        "id": "5d8e2ff1-21bc-4fe0-936f-851407318748",
        "imageUrl": null,
        "authorId": "32df86b7-872d-45d5-8a51-e217f0bc973d",
        "title": "It?",
        "description": null,
        "date": 1615046988090,
        "category": "IT"
      },
      {
        "id": "5b28e638-ef12-45a7-b629-84350cc0c92e",
        "imageUrl": null,
        "authorId": "5d48bb9e-f1d9-41d2-bb02-9c4f578bd282",
        "title": "Bio?",
        "description": null,
        "date": 1615047293904,
        "category": "IT"
      }
    ],
    "avatarLoadingStatus": "LOADED",
    "detailsLoadingStatus": "LOADED",
    "subsLoadingStatus": "LOADED",
    "blackListLoadingStatus": "LOADED",
    "likedListLoadingStatus": "LOADED",
    "feedLoadingStatus": "LOADING",
    "feed": {
      "articles": [],
      "accounts": null
    },
    "myArticles": [
      {
        "id": "9064e587-3c29-4547-a5a6-51a8149456f8",
        "imageUrl": null,
        "authorId": "5d48bb9e-f1d9-41d2-bb02-9c4f578bd282",
        "title": "Русские мемчанские",
        "description": null,
        "date": 1615046988090,
        "category": "Anime"
      },
      {
        "id": "5b28e638-ef12-45a7-b629-84350cc0c92e",
        "imageUrl": null,
        "authorId": "5d48bb9e-f1d9-41d2-bb02-9c4f578bd282",
        "title": "Bio?",
        "description": null,
        "date": 1615047293904,
        "category": "IT"
      }
    ],
    "myArticlesLoadingStatus": "LOADED"
  },
  "articleReducer": {
    "articleDataById": {
      "d3268ef9-bb52-4967-b1ee-14826b9d8a18": {
        "id": "d3268ef9-bb52-4967-b1ee-14826b9d8a18",
        "imageUrl": null,
        "description": null,
        "authorId": "22bb3fe5-185c-4c87-8633-1586586506b7",
        "title": "Chem?",
        "date": 1615046988090,
        "category": "Chemistry",
        "data": "yes, and Jews are responsible."
      },
      "5d8e2ff1-21bc-4fe0-936f-851407318748": {
        "id": "5d8e2ff1-21bc-4fe0-936f-851407318748",
        "imageUrl": null,
        "description": null,
        "authorId": "32df86b7-872d-45d5-8a51-e217f0bc973d",
        "title": "It?",
        "date": 1615046988090,
        "category": "IT",
        "data": "yes, and Jews are responsible."
      },
      "9064e587-3c29-4547-a5a6-51a8149456f8": {
        "id": "9064e587-3c29-4547-a5a6-51a8149456f8",
        "imageUrl": null,
        "description": null,
        "authorId": "5d48bb9e-f1d9-41d2-bb02-9c4f578bd282",
        "title": "Русские мемчанские",
        "date": 1615046988090,
        "category": "Anime",
        "data": "yes, and Jews are responsible."
      },
      "5b28e638-ef12-45a7-b629-84350cc0c92e": {
        "id": "5b28e638-ef12-45a7-b629-84350cc0c92e",
        "imageUrl": null,
        "description": null,
        "authorId": "5d48bb9e-f1d9-41d2-bb02-9c4f578bd282",
        "title": "Bio?",
        "date": 1615047293904,
        "category": "IT",
        "data": "<div><p>Рассмотрим две <b>штуки</b></p><p>Первое:</p><p><b><br></b></p></div>"
      }
    },
    "articleIdsByAuthorName": {}
  },
  "allUsersReducer": {
    "userDataById": {
      "22bb3fe5-185c-4c87-8633-1586586506b7": {
        "accountId": "22bb3fe5-185c-4c87-8633-1586586506b7",
        "name": null,
        "bio": "populated",
        "gender": "gender",
        "dob": 1615046987793,
        "avatar": "blob:http://localhost:3000/595ee9e0-fbd4-4712-98e1-1fd759dca517"
      },
      "32df86b7-872d-45d5-8a51-e217f0bc973d": {
        "accountId": "32df86b7-872d-45d5-8a51-e217f0bc973d",
        "name": null,
        "bio": "populated",
        "gender": "gender",
        "dob": 1615046987793,
        "avatar": "blob:http://localhost:3000/2bd315a7-60e0-41b7-b856-e48559d16d73"
      },
      "5d48bb9e-f1d9-41d2-bb02-9c4f578bd282": {
        "accountId": "5d48bb9e-f1d9-41d2-bb02-9c4f578bd282",
        "name": "Круто чел",
        "bio": "populated",
        "gender": "gender",
        "dob": 1615046987793,
        "avatar": "blob:http://localhost:3000/678d4330-3d93-4470-98f3-4619ed42c932"
      }
    }
  },
  "searchReducer": {
    "articles": [],
    "authors": []
  }
}
