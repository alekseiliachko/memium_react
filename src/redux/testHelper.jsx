import {Provider} from "react-redux";
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import React from "react";

export const getStubStoreData =  () => require('./testData.json');

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

export const withStore = (initialStore) => {
    const store = mockStore(initialStore || getStubStoreData());
    return function Wrapper({ children }) {
        return <Provider store={store}>
            {children}
        </Provider>
    }
}
