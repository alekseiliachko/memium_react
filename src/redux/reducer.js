import { combineReducers, createStore, applyMiddleware, compose } from "redux";
import thunk from "redux-thunk";
import { authReducer } from "./auth/reducer";
import { userReducer } from "./user/reducer";
import { articleReducer } from "./article/reducer";
import { allUsersReducer } from "./allUsers/reducer";
import { searchReducer } from "./search/reducer";

const rootState = combineReducers({
  authReducer,
  userReducer,
  articleReducer,
  allUsersReducer,
  searchReducer,
});

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
const middleware = [thunk];
export function configureStore() {
  const store = createStore(
    rootState,
    composeEnhancers(applyMiddleware(...middleware))
  );
  return store;
}
