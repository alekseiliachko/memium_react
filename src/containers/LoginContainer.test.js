import React from "react";
import renderer from "react-test-renderer";
import { withStore } from "../redux/testHelper";
import { LoginFormContainer } from "./LoginContainer";
import { MemoryRouter } from "react-router";

describe("login container", () => {
  it("renders correctly", async () => {
    const Comp = withStore();
    let tree;

    await renderer.act(async () => {
      tree = renderer.create(
        <Comp>
          <MemoryRouter>
            <LoginFormContainer />
          </MemoryRouter>
        </Comp>
      );
      await Promise.resolve();
    });

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
