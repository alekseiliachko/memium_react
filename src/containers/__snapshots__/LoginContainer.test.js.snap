// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login container renders correctly 1`] = `
<div
  className="MuiContainer-root login-form MuiContainer-maxWidthXs"
  onSubmit={[Function]}
>
  <div
    className="MuiBox-root MuiBox-root-1"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6h2c0-1.66 1.34-3 3-3s3 1.34 3 3v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm0 12H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
      />
    </svg>
  </div>
  <h1
    className="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter"
  >
    Log in
  </h1>
  <form>
    <div
      className="MuiBox-root MuiBox-root-2"
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
          data-shrink={false}
        >
          Username
          <span
            aria-hidden={true}
            className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input"
            disabled={false}
            name="username"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="text"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-3 MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="PrivateNotchedOutline-legendLabelled-5"
            >
              <span>
                Username
                 *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      className="MuiBox-root MuiBox-root-7"
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
          data-shrink={false}
        >
          Password
          <span
            aria-hidden={true}
            className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input"
            disabled={false}
            name="password"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="password"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-3 MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="PrivateNotchedOutline-legendLabelled-5"
            >
              <span>
                Password
                 *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge MuiButton-fullWidth"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="submit"
    >
      <span
        className="MuiButton-label"
      >
        Log in
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
    <div
      className="MuiBox-root MuiBox-root-8"
    >
      <a
        href="/forgot-password"
        onClick={[Function]}
      >
        Забыли пароль?
      </a>
      <a
        href="/signup"
        onClick={[Function]}
      >
        Зарегистрироваться
      </a>
    </div>
  </form>
</div>
`;
